Assets/MRTK/SDK/Features/UX/Scripts/Slate/HandInteractionPanZoom.cs

```diff
@@ -379,46 +379,69 @@ private void UpdateIdle()
 
         private void UpdateUVMapping()
         {
-            Vector2 tiling = currentMaterial != null ? currentMaterial.mainTextureScale : new Vector2(1.0f, 1.0f);
-            Vector2 uvTestValue;
             mesh.GetUVs(0, uvs);
             uvsOrig.Clear();
             uvsOrig.AddRange(uvs);
-            float scaleUVDelta = 0.0f;
-            Vector2 scaleUVCentroid = Vector2.zero;
-            float currentContactRatio = 0.0f;
 
+            // Scale
             if (ScaleActive)
             {
-                scaleUVCentroid = GetDisplayedUVCentroid(uvs);
-                currentContactRatio = GetUVScaleFromTouches();
-                scaleUVDelta = currentContactRatio / previousContactRatio;
+                var scaleUVCentroid = GetDisplayedUVCentroid(uvs);
+                var currentContactRatio = GetUVScaleFromTouches();
+                var scaleUVDelta = currentContactRatio / previousContactRatio;
                 previousContactRatio = currentContactRatio;
 
                 currentScale = totalUVScale.x / scaleUVDelta;
 
                 // Test for scale limits
                 if (currentScale > minScale && currentScale < maxScale)
                 {
-                    // Track total scale
-                    totalUVScale /= scaleUVDelta;
-                    for (int i = 0; i < uvs.Count; ++i)
+                    var scaleUVDeltas = new List<Vector2>();
+                    for (int i = 0; i < uvs.Count; i++)
                     {
-                        // This is where zoom is applied if Active
-                        uvs[i] = ((uvs[i] - scaleUVCentroid) / scaleUVDelta) + scaleUVCentroid;
+                        Vector2 delta = ((uvs[i] - scaleUVCentroid) / scaleUVDelta) + scaleUVCentroid - uvs[i];
+                        scaleUVDeltas.Add(delta);
                     }
+                    UpdateUV(uvs, scaleUVDeltas);
```

Comment:
To double check, if we end up in this `if`, it looks like we call some sort of `UpdateUV` method twice:

1. `UpdateUV` here
2. `UpdateUVWithScroll` on line 417

Do we need to call them both?

Reply:
I guess if you're scaling _and_ scrolling at the same time?

Reply:
Yeah it seems like we can do some logic juggling to make this code cleaner and avoid making multiple calls.

Going to open a new PR soon to push these larger scale changes.